be/src/exec/vectorized/hash_join_node.cpp

```diff
@@ -471,12 +461,18 @@ pipeline::OpFactories HashJoinNode::decompose_to_pipeline(pipeline::PipelineBuil
         }
     }
 
+    size_t num_right_partitions = down_cast<SourceOperatorFactory*>(rhs_operators[0].get())->degree_of_parallelism();
+    size_t num_left_partitions = down_cast<SourceOperatorFactory*>(lhs_operators[0].get())->degree_of_parallelism();
+    // For non-broadcast join, the number of left and right partitions must be the same.
+    DCHECK(_distribution_mode == TJoinDistributionMode::BROADCAST || num_right_partitions == num_left_partitions);
+
```

Comment:
need to throw or return errors?

Reply:
Sounds reasonable. I would make ` ExecNode::decompose_to_pipeline` be able to return errors in the next PR, since it need lots of work to modify codes for each callee of `decompose_to_pipeline`.